src/Core/Component.js

```diff
@@ -65,50 +53,91 @@ function forceUpdate (callback) {
 	enqueueComponentUpdate(getComponentElement(this), this, callback, SharedComponentForceUpdate)
 }
 
+/**
+ * @param {object} description
+ * @return {function}
+ */
+function createClass (description) {
+	return createComponentClass(Object(description), getDisplayName(description))
+}
+
+/**
+ * @param {object} description
+ * @param {string} displayName
+ * @return {function}
+ */
+function createComponentClass (description, displayName) {
+	switch (typeof description) {
+		case 'function':
+			return description[SymbolComponent] = createComponentClass(merge({render: description}, description))
+		case 'object':
+			for (var name in description)
+				description[name] = {value: description[name]}
+	}
+
+	function klass () {}
+
+	klass[SharedSitePrototype] = create(Component[SharedSitePrototype], description)
+	klass[SharedSiteDisplayName] = displayName
+
+	return klass
+}
+
+/**
+ * @param {Object} prototype
+ * @return {Object}
+ */
+function createComponentPrototype (prototype) {
+	defineProperty(prototype, SymbolComponent, {value: SymbolComponent})
+	defineProperty(prototype, SharedSiteSetState, {value: setState})
+	defineProperty(prototype, SharedSiteForceUpdate, {value: forceUpdate})
+
+	if (!prototype[SharedSiteRender])
+		defineProperty(prototype, SharedSiteRender, {value: noop, configurable: true})
+
+	return prototype
+}
+
 /**
  * @param {Element} element
  */
 function mountComponentElement (element) {
-	var owner = element.type
+	var type = element.type
 	var props = element.props
-	var context = element.context || {}
-	var prototype = owner.prototype
-	var instance
+	var host = Object(element.host)
+	var context = element.context = Object(host.context)
+	var prototype = type[SharedSitePrototype]
 	var children
 	var state
+	var owner
 
-	if (prototype && prototype.render) {
-		if (prototype[SymbolComponent] !== SymbolComponent)
-			createComponent(prototype)
-
-		instance = owner = getComponentInstance(element, owner)
-	} else {
-		defineProperty(instance = new Component(), SharedSiteRender, {value: owner, writable: true})
-	}
+	if (prototype && prototype.render)
+		!prototype[SymbolComponent] && createComponentPrototype(prototype)
+	else
+		type = type[SymbolComponent] || createComponentClass(type, getDisplayName(type))
 
-	element.owner = owner
-	element.instance = instance
-	element.context = context
+	element.owner = owner = getLifecycleClass(element, type, props, context)
+	owner[SymbolContext] = element.xmlns = host.xmlns
+	owner[SymbolElement] = element
+	owner.props = props
+	owner.context = context
 
-	instance[SymbolElement] = element
-	instance.refs = {}
-	instance.props = props
-	instance.context = context
-	instance.state = state = instance.state || {}
+	owner.state = state = Object(owner.state)
```

Comment:
`Object()` wrapper here seems redundant due to the defaults in the `Component` class (line 6)

Reply:
This prevents the following from resulting in a non-object state reference

```js
class A extends Component {
constructor() {
this.state = getSomeValueThatMightBeNull()
}
}
```